### ISSUE TITLE
Final double is missing

### ISSUE THREAD

**Issue by ThorvaldAagaard** [lorserker/ben#41]
During test of #38 I once missclicked in the bidding resulting in this sequence
![image](https://github.com/lorserker/ben/assets/10560652/b11bb872-5203-4093-9e49-7a213aabb14f)

I know this sequence will not be present in the training dataset, so it is understandable that BEN will be in trouble.

Perhaps East expects West to double, but West is making the final decision and here it seems only Pass is a possible bid:
![image](https://github.com/lorserker/ben/assets/10560652/0956fb71-30c6-4f9c-b269-27c1e8948c29)
The simulated hands for West does not match the bidding very good, but the real problem seems to be that Double isn't evaluated

**Comment by lorserker** [comment]
yes, i have seen this bug in other situations too.
the problem is that the model suggests pass so strongly that nothing else is evaluated. if double was evaluated, it very likely would be selected.

**Comment by lorserker** [comment]
i think it is a possible solution to add some heuristic to force consideration of double if the opponents are known to be much weaker or in other situations to consider more bids if it looks like we end up in a contract with few trumps.

**Comment by ThorvaldAagaard** [comment]
I think that is a good idea.

I just tested a known bug from GIB, where you as opponent bid 4N over an 2Club opening. It resulted in this:
![image](https://github.com/lorserker/ben/assets/10560652/df02599e-8597-45dc-8b52-d12fb1a39b03)

I am still a bit confused about the randomness of the bids, as when I reran this board North now bid 5D (And now East doubled)

Is there some random seed I can set, so I always get the same output on a given input?

**Comment by lorserker** [comment]
@ThorvaldAagaard the randomness comes from the sampling.
it is possible to seed the random number generator, but unfortunately I didn't do it when I wrote the code.

**Comment by ThorvaldAagaard** [comment]
Adding

```
np.random.seed(42)
```

in the Driver Class init method in game.py seems to fix the problem.

![image](https://github.com/lorserker/ben/assets/10560652/24568c3a-d0aa-488a-9dc5-686d731ac212)

I can see that there is a mix of use of random and np.random, so I also changed all use of random to np.random (Could also seed random in the Driver Class, but I think it is better just to use one random-function)

I will take a look at how find patterns, where a Double, Redouble or Pass should be considered when not suggested by the Neural network

**Comment by ThorvaldAagaard** [comment]
I have added a principle of lowering the threshold in the pass out situation, so BEN will now double 3N, and also find a double after the 2C-opening.

Interesting if we will see many more penalty doubles.
